spring-web/src/main/java/akka/stream/alpakka/spring/web/AkkaStreamsRegistrar.java

```diff
@@ -25,15 +26,16 @@ public void registerAdapters(ReactiveAdapterRegistry registry) {
         multiValue(akka.stream.javadsl.Source.class, akka.stream.javadsl.Source::empty),
         source ->
             ((akka.stream.javadsl.Source<?, ?>) source)
-                .runWith(
-                    akka.stream.javadsl.Sink.asPublisher(AsPublisher.WITH_FANOUT), materializer),
+                .runWith(akka.stream.javadsl.Sink.asPublisher(AsPublisher.WITH_FANOUT), system),
         akka.stream.javadsl.Source::fromPublisher);
 
     registry.registerReactiveType(
         multiValue(akka.stream.scaladsl.Source.class, akka.stream.scaladsl.Source::empty),
         source ->
             ((akka.stream.scaladsl.Source<?, ?>) source)
-                .runWith(akka.stream.scaladsl.Sink.asPublisher(true), materializer),
+                .runWith(
+                    akka.stream.scaladsl.Sink.asPublisher(true),
+                    Materializer.matFromSystem(system)),
```

Comment:
Shouldn't this be just `system`?

Reply:
sadly, I have the compile-time error in that case :( Perhaps, I should do the relevant change in akka-streams.

Reply:
No there are no changes required. The problem here is that it is using the Scala DSL from Java code. Your explicit use of `matFromSystem` is correct.

Reply:
yes, indeed. 
However, the logic does require scala dsl. Hence, my thought on having the corresponding method (accepting `system`, not `materializer`)

Reply:
I can wrap the scaladsl.Source with the javadsl.Source.
Besides, the constructor may look as follows 

```java
    /**
     * @deprecated use {@link #AkkaStreamsRegistrar(ActorSystem)}.
     */
  @Deprecated
  public AkkaStreamsRegistrar(Materializer materializer) {
      this(materializer.system());
  }
```

Shall I ?